hack/components/common_functions.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -xe
+
+function fetch_component() {
+    destination=$1
+    url=$2
+    commit=$3
+
+    if [ ! -d ${destination} ]; then
+        mkdir -p ${destination}
+        git clone ${url} ${destination}
+    fi
+
+    (
```

Comment:
why sometimes you open this scope ?

Reply:
this is not a scope, the `()` opens a subshell that performs all the commands in it.

Reply:
Is about keeping path, since it opens subshell after it finished we will keep the previous path

Reply:
thanks, what about pushd popd ?